framework/base/Model.php

```diff
@@ -917,9 +929,15 @@ public static function validateMultiple($models, $attributeNames = null)
      */
     public function fields()
     {
-        $fields = $this->attributes();
+        static $fields = null;
+
+        if ($fields === null) {
+            $attributes = $this->attributes();
```

Comment:
Ah, you're right, I didn't consider someone overriding Model.attributes(), I'll have to revert this one. (.. continued below)